File: laboratory/369.py
def get_user_input():
    print('Please input the last number')
    number = int(input())

    return number


def print_369_until(number):
    print('-' * 20)

    # 1 부터 number까지
    for i in range(1, number + 1):
        is_contains_369 = False

        # 숫자를 문자열로 변환 후 한자리씩 리턴
        for j in str(i):
            # 3, 6, 9가 j안에 포함되어있는 경우
            if j in ['3', '6', '9']:
                print('clap ', end='')
                is_contains_369 = True

        # 3, 6, 9가 포함 안 된 경우만 숫자 출력
        if not is_contains_369:
            print(i)
        else:
            print()


def main():
    number = get_user_input()

    print_369_until(number)


if __name__ == '__main__':
    main()

File: laboratory/369.c
#include <stdio.h>

int main(void) {
	int i, j;
	int number;

	printf("Please input the last number\n");
	scanf("%d", &number);

	printf("-------- start --------\n");

	for (i = 1; i <= number; i++) {
		int clapCount = 0;
		// i 값을 calcNum에 복사
		int calcNum = i;

		while (calcNum > 0) {
			// 1의 자릿수
			int extra = calcNum % 10;

			// 3, 6, 9가 포함된 경우
			if (extra == 3 || extra == 6 || extra == 9) {
				// 박수 카운트 증가
				clapCount++;
			}

			calcNum = calcNum / 10;
		}

		// 박수 칠 카운트가 있는 경우
		if (clapCount > 0) {
			for (j = 0; j < clapCount; j++) {
				printf("clap ");
			}
			printf("\n");
		} else {
			printf("%d \n", i);
		}
	}

	return 0;
}